packages/contracts/scripts/util/utils.ts

```diff
@@ -1,12 +1,50 @@
 import prompt from "prompt";
+import hre from "hardhat";
 import fs from "fs";
 import path from "path";
+import { ethers } from "ethers";
 var util = require("util");
 
 export function delay(ms: number) {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
 
+/**
+ * Gets the configured max nominal gas price in gwei for the current network
+ * @returns the max nominal gas price in gwei for the current network
+ */
+function getMaxNominalGasPriceGwei(): number {
+  // Require maxNominalGasPrice defined in hardhat config
+  const maxNominalGasPriceGwei: number =
+    hre.network.config.maxNominalGasPriceGwei;
+  if (!maxNominalGasPriceGwei) {
+    throw new Error(
+      `[ERROR] maxNominalGasPrice must be defined in hardhat.config.ts for the current network ${hre.network.name}`
+    );
+  }
+  return maxNominalGasPriceGwei;
+}
+
+/**
+ * Ensures the current gas price is below the configured max nominal gas price
+ * Throws an error if the current gas price is above the configured max nominal gas price
+ */
+export async function requireLowGasPrice() {
+  const maxNominalGasPriceGwei = getMaxNominalGasPriceGwei();
+  // get gas price and ensure it is below the max nominal gas price
+  const gasPriceGwei = parseFloat(
+    ethers.utils.formatUnits(await hre.ethers.provider.getGasPrice(), "gwei")
+  );
+  if (gasPriceGwei > maxNominalGasPriceGwei) {
+    throw new Error(
+      `[ERROR] Gas price is ${gasPriceGwei.toString()} gwei, which is above the configured max nominal gas price of ${maxNominalGasPriceGwei} gwei`
+    );
+  }
+  console.log(
+    `[INFO] Gas price is ${gasPriceGwei.toString()} gwei, which is lt the configured max nominal gas price of ${maxNominalGasPriceGwei} gwei`
+  );
+}
```

Comment:
Nit: I would think the only place we anticipate this to be used is in `packages/contracts/scripts/require-low-gas.ts` itself, would suggest that we move it there for now and can split off into utils later if actually needed?

Reply:
good call, updated!